ha/monitor/k8s/parser.py

```diff
@@ -156,13 +156,17 @@ def _create_health_alert(self, res_type, res_name, health_status, generation_id)
         self.event.set_specific_info({"generation_id": generation_id})
         return self.event.json
 
-    def parse(self, an_event, cached_state):
+    def parse(self, an_event, cached_state, resource_id_map):
         resource_type = self._type
         raw_object = an_event[K8SEventsConst.RAW_OBJECT]
 
         labels = raw_object[K8SEventsConst.METADATA][K8SEventsConst.LABELS]
-        if K8SEventsConst.MACHINEID in labels:
-            resource_name = labels[K8SEventsConst.MACHINEID]
+        if K8SEventsConst.LABEL_PODNAME in labels:
+            resource_name = resource_id_map.get(labels[K8SEventsConst.LABEL_PODNAME], None)
```

Comment:
I think variable name should be resource_id. resource_name is bit confusing. Its giving the impression that its an actual resource_name(here, pod_name).

Reply:
Also is this only needed for pod resources or for node (physical resources) as well though I think we were not consuming those yet.

Reply:
I have chnages it to resource_id, but the reason why I was not changed it is: that this variable name is already there However we are using resource id earlier also, and it is in the whole function, but now changed let's not make more late.
But I have changed the variable name for the node event parser as we are using the node name as it is and I think, this answers @shaileshsvc  question too.

Reply:
So, are you keeping it as it is or changing it?

Reply:
If you have verified that there is no issues with node events filtering in k8smonitor due to fqdn change, please consider my comment resolved. Thanks.

Reply:
@Madhura-08 
> So, are you keeping it as it is or changing it?
I have change it as per your comment for pod event parser

Reply:
> If you have verified that there is no issues with node events filtering in k8smonitor due to fqdn change, please consider my comment resolved. Thanks.

for node event parser node name used in alert as resource id.
in code you can see we are using node name:
```
resource_name = raw_object[K8SEventsConst.METADATA][K8SEventsConst.NAME]
```

Reply:
@Madhura-08 its already done please check